# Carga y revisión del dataset

En este notebook se realiza la carga y exploración inicial del dataset limpio de TelecomX.

Objetivos:

- Cargar el dataset preparado en el challenge anterior
- Revisar la estructura del dataset
- Analizar la variable objetivo (Churn)
- Identificar variables numéricas y categóricas
- Detectar posibles problemas antes del modelado

##Montamos Drive

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


##Importar las librerías necesarias

In [5]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

sns.set(style="whitegrid")

##Definir la ruta del dataset

In [6]:
DATA_PATH = "/content/drive/MyDrive/TelecomX_Churn_ML/data/processed/TelecomX_clean.csv"

##Cargamos el dataset que obtuvimos en el anterior Challenge

In [7]:
df = pd.read_csv(DATA_PATH)

df.head()

,id_cliente,abandono,genero,adulto_mayor,tiene_pareja,tiene_dependientes,antiguedad_meses,servicio_telefono,lineas_multiples,servicio_internet,...,proteccion_dispositivo,soporte_tecnico,streaming_tv,streaming_peliculas,tipo_contrato,factura_digital,metodo_pago,cargo_mensual,cargo_total,cargo_diario
0,0002-ORFBO,0,Female,0,1,1,9,1,0,DSL,...,0,1,1,0,One year,1,Mailed check,65.6,593.30,2.186667
1,0003-MKNFE,0,Male,0,0,0,9,1,1,DSL,...,0,0,0,1,Month-to-month,0,Mailed check,59.9,542.40,1.996667
2,0004-TLHLJ,1,Male,0,0,0,4,1,0,Fiber optic,...,1,0,0,0,Month-to-month,1,Electronic check,73.9,280.85,2.463333
3,0011-IGKFF,1,Male,1,1,0,13,1,0,Fiber optic,...,1,0,1,1,Month-to-month,1,Electronic check,98.0,1237.85,3.266667
4,0013-EXCHZ,1,Female,1,1,0,3,1,0,Fiber optic,...,0,1,1,0,Month-to-month,1,Mailed check,83.9,267.40,2.796667


##Visualizamos las dimensiones del dataset

In [8]:
print("Número de filas:", df.shape[0])
print("Número de columnas:", df.shape[1])

Número de filas: 7032
Número de columnas: 22


##Información del dataset

In [10]:
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7032 entries, 0 to 7031
Data columns (total 22 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id_cliente              7032 non-null   object 
 1   abandono                7032 non-null   int64  
 2   genero                  7032 non-null   object 
 3   adulto_mayor            7032 non-null   int64  
 4   tiene_pareja            7032 non-null   int64  
 5   tiene_dependientes      7032 non-null   int64  
 6   antiguedad_meses        7032 non-null   int64  
 7   servicio_telefono       7032 non-null   int64  
 8   lineas_multiples        7032 non-null   int64  
 9   servicio_internet       7032 non-null   object 
 10  seguridad_online        7032 non-null   object 
 11  respaldo_online         7032 non-null   object 
 12  proteccion_dispositivo  7032 non-null   int64  
 13  soporte_tecnico         7032 non-null   int64  
 14  streaming_tv            7032 non-null   

,abandono,adulto_mayor,tiene_pareja,tiene_dependientes,antiguedad_meses,servicio_telefono,lineas_multiples,proteccion_dispositivo,soporte_tecnico,streaming_tv,streaming_peliculas,factura_digital,cargo_mensual,cargo_total,cargo_diario
count,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000,7032.000000
mean,0.265785,0.162400,0.482509,0.298493,32.421786,0.903299,0.421928,0.343857,0.290102,0.384386,0.388367,0.592719,64.798208,2283.300441,2.159940
std,0.441782,0.368844,0.499729,0.457629,24.545260,0.295571,0.493902,0.475028,0.453842,0.486484,0.487414,0.491363,30.085974,2266.771362,1.002866
min,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,18.250000,18.800000,0.608333
25%,0.000000,0.000000,0.000000,0.000000,9.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,35.587500,401.450000,1.186250
50%,0.000000,0.000000,0.000000,0.000000,29.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,70.350000,1397.475000,2.345000
75%,1.000000,0.000000,1.000000,1.000000,55.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,89.862500,3794.737500,2.995417
max,1.000000,1.000000,1.000000,1.000000,72.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,118.750000,8684.800000,3.958333


##Nombres de las columnas del dataset

In [11]:
df.columns

Index(['id_cliente', 'abandono', 'genero', 'adulto_mayor', 'tiene_pareja',
       'tiene_dependientes', 'antiguedad_meses', 'servicio_telefono',
       'lineas_multiples', 'servicio_internet', 'seguridad_online',
       'respaldo_online', 'proteccion_dispositivo', 'soporte_tecnico',
       'streaming_tv', 'streaming_peliculas', 'tipo_contrato',
       'factura_digital', 'metodo_pago', 'cargo_mensual', 'cargo_total',
       'cargo_diario'],
      dtype='object')

##Revisión de la variable objetivo

In [14]:
df["abandono"].value_counts()



,count
abandono,
0,5163
1,1869


In [15]:
df["abandono"].value_counts(normalize=True)*100

,proportion
abandono,
0,73.421502
1,26.578498


### Distribución de la variable objetivo

La variable `abandono` muestra que aproximadamente el 26.6% de los clientes cancelaron el servicio, mientras que el 73.4% permanecen activos. Esto indica un desbalance moderado entre clases.

Este comportamiento es consistente con el análisis exploratorio realizado previamente y deberá considerarse durante la evaluación de los modelos, priorizando métricas como recall y F1-score para la detección de clientes con riesgo de cancelación.

## Eliminación de columnas no predictivas

La columna `id_cliente` corresponde a un identificador único de cada cliente.  
Este tipo de variable no aporta valor al modelo predictivo y puede afectar el desempeño del algoritmo.

Por esta razón será eliminada del dataset antes del proceso de modelado.

In [16]:
df = df.drop(columns=["id_cliente"])

df.head()

,abandono,genero,adulto_mayor,tiene_pareja,tiene_dependientes,antiguedad_meses,servicio_telefono,lineas_multiples,servicio_internet,seguridad_online,...,proteccion_dispositivo,soporte_tecnico,streaming_tv,streaming_peliculas,tipo_contrato,factura_digital,metodo_pago,cargo_mensual,cargo_total,cargo_diario
0,0,Female,0,1,1,9,1,0,DSL,No,...,0,1,1,0,One year,1,Mailed check,65.6,593.30,2.186667
1,0,Male,0,0,0,9,1,1,DSL,No,...,0,0,0,1,Month-to-month,0,Mailed check,59.9,542.40,1.996667
2,1,Male,0,0,0,4,1,0,Fiber optic,No,...,1,0,0,0,Month-to-month,1,Electronic check,73.9,280.85,2.463333
3,1,Male,1,1,0,13,1,0,Fiber optic,No,...,1,0,1,1,Month-to-month,1,Electronic check,98.0,1237.85,3.266667
4,1,Female,1,1,0,3,1,0,Fiber optic,No,...,0,1,1,0,Month-to-month,1,Mailed check,83.9,267.40,2.796667


## Conclusiones de la revisión del dataset

El dataset tratado fue cargado correctamente y contiene 7032 registros con 21 variables.

Se eliminó la columna `id_cliente` ya que corresponde a un identificador único y no aporta información útil para los modelos predictivos.

La variable objetivo del modelo será `abandono`, que indica si el cliente canceló el servicio (1) o permanece activo (0).

El dataset se encuentra listo para iniciar la etapa de preprocesamiento para Machine Learning.
